crates/swift-bridge-ir/src/bridged_type/bridgeable_result.rs

```diff
@@ -220,16 +305,108 @@ impl BuiltInResult {
         }
     }
 
-    pub fn to_c(&self) -> &'static str {
+    pub fn to_c(&self) -> String {
+        if self.is_custom_result_type() {
+            return format!(
+                "struct {}$Result{}And{}",
+                SWIFT_BRIDGE_PREFIX, self.ok_ty_string, self.err_ty_string
+            );
+        }
         // TODO: Choose the kind of Result representation based on whether or not the ok and error
         //  types are primitives.
         //  See `swift-bridge/src/std_bridge/result`
         if self.ok_ty.can_be_encoded_with_zero_bytes() {
-            "struct __private__ResultVoidAndPtr"
+            format!("struct __private__ResultVoidAndPtr")
         } else {
-            "struct __private__ResultPtrAndPtr"
+            format!("struct __private__ResultPtrAndPtr")
         }
     }
+
+    pub fn generate_ffi_definition(
+        &self,
+        swift_bridge_path: &Path,
+        types: &TypeDeclarations,
+    ) -> Option<TokenStream> {
+        if self.is_custom_result_type() {
+            if self.err_ty.can_be_encoded_with_zero_bytes() {
+                todo!()
+            }
+            let ty = self.to_ffi_compatible_rust_type(swift_bridge_path);
+            let ok = if self.ok_ty.can_be_encoded_with_zero_bytes() {
+                quote! {* mut std :: ffi :: c_void}
+            } else {
+                self.ok_ty
+                    .to_ffi_compatible_rust_type(swift_bridge_path, types)
+            };
+
+            let err = self
+                .err_ty
+                .to_ffi_compatible_rust_type(swift_bridge_path, types);
+            return Some(quote! {
+                #[repr(C)]
+                pub enum #ty {
+                    Ok(#ok),
+                    Err(#err),
+                }
+            });
+        }
+        return None;
+    }
+
+    pub fn generate_c_declaration(&self) -> Option<String> {
+        if self.is_custom_result_type() {
```

Comment:
We can make the early return path more clear, as well as remove the indentation here by doing:

```rust
if !self.is_custom_result_type {
    return None;
}

// ...
```